src/NuGet.Core/NuGet.DependencyResolver.Core/Remote/RemoteDependencyWalker.cs

```diff
@@ -118,16 +120,20 @@ public async Task<GraphNode<RemoteResolveResult>> WalkAsync(LibraryRange library
                 }
             }
 
-            var node = new GraphNode<RemoteResolveResult>(libraryRange)
+            // Resolve the dependency from the cache or sources
+            GraphItem<RemoteResolveResult> item = await ResolverUtility.FindLibraryCachedAsync(
+                _context.FindLibraryEntryCache,
+                libraryRange,
+                framework,
+                runtimeName,
+                _context,
+                CancellationToken.None);
+
+            int innerNodesSize = item.Data.Dependencies.Count + (dependencies == null ? 0 : dependencies.Count);
```

Comment:
nit: you can probably just wrap this in a bracket and say > 0.
It's plenty readable without that.

Reply:
Thanks! Fixed.